src/pages/create-product/index.js

```diff
@@ -0,0 +1,170 @@
+import BaseComponent from "../../components/base-component.js";
+import { createProduct, getBrands, getCategories } from "../../api/products.js";
+
+import "./create-product.css";
+
+export default class CreateProductPage extends BaseComponent {
+  constructor() {
+    super();
+    this.createProduct = createProduct;
+    this.getCategories = getCategories;
+    this.getBrands = getBrands;
+
+    this.init();
+    this.initEventListeners();
+    this.loadData();
+  }
+
+  async loadData() {
+    try {
+      const [categories, brands] = await Promise.all([
+        this.getCategories(),
+        this.getBrands(),
+      ]);
+
+      this.update(categories, brands);
+    } catch (error) {
+      this.showAlert("error", error.message);
+    }
+  }
+
+  update(categories = [], brands = []) {
+    this.updateOptions("brandsSelect", brands);
+    this.updateOptions("categoriesSelect", categories);
+
+    this.subElements.fieldset.removeAttribute("disabled");
+  }
+
+  updateOptions(name = "", data = []) {
+    const fragment = new DocumentFragment();
+
+    for (const brand of data) {
+      const key = brand.toLowerCase().split(" ").join("_");
+      const option = new Option(brand, key);
+      fragment.append(option);
+    }
+
+    this.subElements[name].replaceChildren(fragment);
+  }
+
+  startLoading() {
+    this.subElements.fieldset.setAttribute("disabled", "true");
+    this.subElements.submitBtn.innerHTML = `
+      <span class="spinner-border spinner-border-sm" aria-hidden="true"></span>
+      <span role="status">Loading...</span>
+    `;
+  }
```

Comment:
**Scalability Issue**: The startLoading method directly manipulates the innerHTML of elements, which can be inefficient and pose scalability issues when dealing with complex states or a large number of elements. <br> **Fix**: Optimize the startLoading method by using more efficient DOM manipulation techniques or by toggling visibility of pre-existing elements instead of setting innerHTML directly. <br> **Code Suggestion**: 
 ```
 -    this.subElements.submitBtn.innerHTML = '
 -      <span class="spinner-border spinner-border-sm" aria-hidden="true"></span>
 -      <span role="status">Loading...</span>
 -    ';
 +    this.subElements.spinner.classList.remove('d-none');
 +    this.subElements.statusLabel.textContent = 'Loading...';
 ```